drivers/sensor/tmp112/tmp112.c

```diff
@@ -14,32 +14,28 @@
 #include <drivers/sensor.h>
 #include <sys/__assert.h>
 #include <logging/log.h>
+#include "tmp112.h"
 
 LOG_MODULE_REGISTER(TMP112, CONFIG_SENSOR_LOG_LEVEL);
 
-#define TMP112_I2C_ADDRESS		DT_INST_REG_ADDR(0)
-
-#define TMP112_REG_TEMPERATURE		0x00
-#define TMP112_D0_BIT			BIT(0)
-
-#define TMP112_REG_CONFIG		0x01
-#define TMP112_EM_BIT			BIT(4)
-#define TMP112_CR0_BIT			BIT(6)
-#define TMP112_CR1_BIT			BIT(7)
-
-/* scale in micro degrees Celsius */
-#define TMP112_TEMP_SCALE		62500
+static inline const struct tmp112_config *get_config(const struct device *dev)
+{
+	return dev->config;
+}
```

Comment:
This function is redundant, `struct tmp112_config *config = dev->config` can be used instead

Reply:
I used to have compilation warning when doing so let me check again

Reply:
I'm not against the usage of inline functions like this, honestly. The code looks more clear and compiler should optimize anyway. But if you want to follow the suggestion I think the warning issue is maybe because the "const" dropped.

Reply:
I also like those accessor function. they are present in some other device drivers as well

Reply:
I think most of these helpers come from a period when Zephyr had _complicated names_ for the config/data fields in the device struct. I won't block on that, but I think new or refactored drivers should stop using such dummy functions.